torch/nested/__init__.py

```diff
@@ -302,3 +304,87 @@ def narrow(tensor: Tensor, dim: int, start: Union[int, Tensor], length: Union[in
         raise RuntimeError(f"Specified layout is unsupported for nested narrow: {layout}")
 
     return nt
+
+
+def nested_tensor_from_jagged(
+    values: Tensor,
+    offsets: Optional[Tensor] = None,
+    lengths: Optional[Tensor] = None,
+    jagged_dim: Optional[int] = None,
```

Comment:
Some might also want to pass the `size` if this is a reconstruction after a decomposition via `.offsets()` and `.values()`.